File: 1984.minimum-difference-between-highest-and-lowest-of-k-scores.go
/*
 * @lc app=leetcode id=1984 lang=golang
 *
 * [1984] Minimum Difference Between Highest and Lowest of K Scores
 */

package main

import (
	"sort"
)

// @lc code=start
func minimumDifference(nums []int, k int) int {
	if len(nums) == 1 {
		return 0
	}

	sort.Ints(nums)

	var ret int = 1e5 + 1
	for i := 0; i+k <= len(nums); i++ {
		diff := nums[i] - nums[i+k-1]
		if diff < 0 {
			diff *= -1
		}
		ret = min(diff, ret)
	}
	return ret
}

// @lc code=end

File: 1984.minimum-difference-between-highest-and-lowest-of-k-scores.rs
/*
 * @lc app=leetcode id=1984 lang=rust
 *
 * [1984] Minimum Difference Between Highest and Lowest of K Scores
 */

// @lc code=start
impl Solution {
    pub fn minimum_difference(mut nums: Vec<i32>, k: i32) -> i32 {
        if (k == 1) {
            return 0;
        }

        nums.sort();

        let mut ret = 100001;
        let k = k as usize;
        let mut i = 0;
        while i + k <= nums.len() {
            ret = std::cmp::min(ret, nums[i + k - 1] - nums[i]);
            i += 1;
        }

        ret
    }
}
// @lc code=end
